File: python/01-Fundamentals/02-fundamentals.py
#1) Countdown
def countdown(num):
    newList = []
    for i in range(num, -1, -1):
        newList.append(i)
    return newList
#print(countdown(5))

#2) Print and Return
def printReturn(arr):
    print(arr[0])
    return arr[1]
#print(printReturn([23,76]))

#3) First Plus Length
def firstPlus(arr):
    return arr[0] + len(arr)
algo3Array = [5,8,4,5,7]
#print(firstPlus(algo3Array))

#4) Values Greater than Second
def printGreaterSum(arr):
    sum = 0
    for i in range(len(arr)):
        if arr[i] > arr[1]:
            sum += 1
    return sum
algo4Array = [1,3,5,7,9,13]
#print(printGreaterSum(algo4Array))

#5) Values Greater than Second, Generalized
def printGreaterSum2(arr):
    if len(arr) < 2:
        return "Array too short."
    sum = 0
    newArray = []
    for i in range(len(arr)):
        if arr[i] > arr[1]:
            sum += 1
            newArray.append(arr[i])
    print(sum)
    return newArray
algo5Array = [1,3,5,7,9,13]
#print(printGreaterSum2(algo5Array))

#6) This Length, That Value
def thisAndThat(num1, num2):
    if num1 == num2:
        return "Jinx!"
    newArray = []
    for i in range(num1):
        newArray.append(num2)
    return newArray
a = 2
b = 12
#print(thisAndThat(a, b))

#7) Fit the First Value
def goldilocks(arr):
    if(arr[0] > len(arr)):
        return "Too big!"
    elif(arr[0] < len(arr)):
        return "Too small!"
    else:
        return "Just right!"
algo7Array = [6,14,53,7,9,13]
#print(goldilocks(algo7Array))

File: javaScript/01-Fundamentals/02-fundamentals.js
'use strict';
//1
function arrayCountdown(input){
    var newArray = [];
    for(var i = input; i >= 0; i--){
        newArray.push(i);
    }
    console.log("My Array is "+newArray.length+" indices long.");
    return newArray;
}
//console.log(arrayCountdown(4));

//2
function printAndReturn(input){
    console.log("Printing: "+input[0]);
    return input[1];
}
//console.log("Returning: "+printAndReturn([32, 44]));

//3
function sumStuff(input){
    return input[0]+input.length;
}
console.log(sumStuff([true, 44]));
// true evaluates as 1+ length
// false evaluates as 0+ length

//4
function return2(input){
    return input[0]+input.length
}
console.log(return2(["what?", 44]));
// string concatonates

//5
// ???

//6
function greaterThan(input, minIndex){
    var sum = 0;
    for(var i = 0; i < input.length; i++){
        if(input[i] > input[minIndex]){
            console.log(input[i]);
            sum += 1;
        }
    }
    return sum;
}
// console.log("Total values greater than "+[1,3,5,7,9,13][1]+" : "+greaterThan([1,3,5,7,9,13], 1));

//7
function greaterThanAny(input, index){
    if(input.length <=1){
        return "need more values than one.";
    }
    var sum = 0;
    for(var i = 0; i < input.length; i++){
        if(input[i] > input[index]){
            console.log(input[i]);
            sum += 1;
        }
    }
    return sum;
}
//console.log("Total values greater than "+[1,3,5,7,9,13][1]+" : "+greaterThanAny([1,3,5,7,9,13], 1));

//8
function thisAndThat(num1, num2){
    if(num1 == num2){
        return "Jinx!";
    }
    var newArray = [];
    for(var i = 0; i <= num1; i++){
        newArray.push(num2);
    }
    return newArray;
}
//console.log(thisAndThat(5, 5));

//9
function goldielocks(input){
    if(input[0] > input.length){
        return "Too big!";
    }
    else if(input[0] < input.length){
        return "Too small!";
    }
    else{
        return "Just right!";
    }
}
//console.log(goldielocks([4,2,3,4]));

//10
function fahrenheitToCelsius(fDegrees){
    var celsius = (fDegrees - 32) / (9/5);
    return celsius;
}
//console.log(fahrenheitToCelsius(14));

//11
function celsiusToFahrenheit(cDegrees){
    var fahrenheit = (9/5 * cDegrees) + 32;
    return fahrenheit;
}
//console.log(celsiusToFahrenheit(-10));